components/right-sidebar/SidebarSocialMedia.tsx

```diff
@@ -1,21 +1,65 @@
-import React from "react";
-import Link from "next/link";
-import SocialMediaLinks from "./SocialMediaLinks";
+import React, { useState } from "react";
 import Button from "../shared/ui/Button";
+import SocialMediaModal from "./SocialMediaModal";
+import { User, Social } from "@prisma/client";
+import { socialMediaIconList } from "@/lib/constants/socialMediaList";
+
+const SidebarSocialMedia = ({
+  user,
+}: {
+  user: User & { socialMedia?: Social[] };
+}) => {
+  const [open, setOpen] = useState(false);
 
-const SidebarSocialMedia = () => {
   return (
     <div className="flex flex-col gap-4">
-      <Link href="/profile/edit/update-socials">
-        <Button icon="plus" color="gray">
-          Update social link
-        </Button>
-      </Link>
-
-      <h3 className="paragraph-2-bold border-white-500 border-b pb-4">
+      <Button icon="plus" color="gray" onClick={() => setOpen(true)}>
+        Update social link
+      </Button>
+      <h3 className="paragraph-2-bold border-white-500 border-b pb-4 pt-4">
         Social Media Links
       </h3>
-      <SocialMediaLinks />
+
+      <div className="flex gap-2">
+        {user.socialMedia ? (
+          user.socialMedia.map((social) => {
+            const filtered = socialMediaIconList.filter(
+              (icon) => social.type === icon.type
+            );
+            const { icon: Icon } = filtered[0];
+
+            return (
+              <div className="flex items-center justify-center">
+                <Icon size={30} />
+                <a
+                  target="_blank"
+                  href={social.link}
+                  className="text-white-300 paragraph-2-regular"
+                >
+                  @{social.username}
+                </a>
+              </div>
+            );
+          })
+        ) : (
+          <p className="text-white-300 paragraph-2-regular">
+            Oops! No socials!
+          </p>
+        )}
+      </div>
+
+      {open && (
+        <div
+          className="relative z-10"
+          aria-labelledby="modal-title"
+          role="dialog"
+          aria-modal="true"
+        >
+          <div className="fixed inset-0 flex items-center justify-center bg-opacity-75 backdrop-blur transition-opacity">
+            <SocialMediaModal user={user} onClose={() => setOpen(false)} />
+          </div>
+        </div>
+      )}
```

Comment:
The only thing I would suggest here would be to use portals to make sure we're add the modal to the end of the DOM. Won't be an issue with your current structure- but you may run into a situation where you're creating a modal inside of sheet or absolute or relatively positioned component and will need to 'break out' of that section of the dom.

React gives us 'createPortal'

https://react.dev/reference/react-dom/createPortal


Which let's us render some component at a specific part of the dom:

```js
{createPortal(<>I'm at the end of the body</>,document.body)}
```

No matter where this code is called- that fragment will be rendered at the end of the DOM. Which is great for modals and making sure things are consistent.

You can simply update your code to use one div instead of a fixed inside of a relative and wrap it in a portal:
```js
      {open &&
        createPortal(
          <div
            aria-labelledby="modal"
            role="dialog"
            aria-modal="true"
            className="fixed inset-0 z-[100] flex items-center justify-center bg-opacity-75 backdrop-blur transition-opacity"
          >
            <SocialMediaModal user={user} onClose={() => setOpen(false)} />
          </div>,
          document.body
        )}
 ```
 
 
![image](https://github.com/DDVVPP/GitNote/assets/4108484/0dd9a419-dadf-4484-ab10-991e1c1ad671)

And we can see it now gets rendered outside of the rest of our code. :)

Here's an example from the portal docs on why this can be useful:
https://react.dev/reference/react-dom/createPortal#rendering-a-modal-dialog-with-a-portal

Reply:
Amazing!